### PR TITLE
Fix url to API doc in readme

### PR DESCRIPTION
**Pull request by jrutgeer**
Changed README to point to the API doc via the rclcpp package info page at the ROS Index.

### CODE REVIEW

**Review by fujitatomoya — COMMENTED** [review]
@jrutgeer thanks for the contribution, this PR surely fixes one link, which is good. but i can see there are many places use `docs.ros2.org`, so those probably need to be changed as well. besides we are having discussion on https://github.com/ros2/ros2/issues/1354.

File: README.md
Comment (ivanpauno) [inline README.md:11]: Maybe link https://docs.ros.org/en/rolling/p/rclcpp/ directly?

```suggestion
Visit the [rclcpp API documentation](https://docs.ros.org/en/rolling/p/rclcpp/) for a complete list of its main components.
```

**Review by ivanpauno — COMMENTED** [review]
what's wrong with the current link?

File: README.md
Comment (jrutgeer) [reply README.md:11]: @ivanpauno I considered that, but a link to the ROS Index has the added benefit of being instructional to new users (that in general API links for packages can be found through the ROS Index).
But that's up to personal opinion...  Main importance is that it's a correct link, as any new user is likely to search for rclcpp API docs before any other package's API, and google isn't helpful.

Comment (clalancette) [reply README.md:11]: Yeah, I have to agree with @jrutgeer here that showing people the general place to find API documentation is a good thing, so I would vote to go with the link to index.ros.org.

**Review by clalancette — APPROVED** [review]
Looks good to me, but I'll wait a bit longer before merging to get other opinions.

**Comment by clalancette** [comment]
We've had no further reviews, so I'm happy to merge it as-is.

@jrutgeer I just realized that the DCO bot is unhappy because the commit does not have the `Signed-off-by:` line.  Can you please rebase, add that, and then force-push it?  Once that is done we can merge.

**Comment by jrutgeer** [comment]
@clalancette I tried to do that, but I think I failed.
Also when I press compare above it seems that 18 files were changed??
I don´t know how come I could have had write access to this repository and not only my local fork?

**Comment by clalancette** [comment]
> I don´t know how come I could have had write access to this repository and not only my local fork?

You only have access to your local fork.  Because you did a merge, you brought in changes that happened since your fork.

In this case, the easiest thing to do is to just make a brand-new commit with a Signed-off-by line, and then open a new PR.

**Pull request closed without merging**
